### DISCUSSION TITLE
「重学TS 2.0 」TS 面试题：TypeScript 的装饰器是什么？它们有几种分类及应用场景？

### DISCUSSION THREAD

**Discussion by semlinker** [semlinker/awesome-typescript#82] [General]
### TypeScript 的装饰器是什么？它们有几种分类及应用场景？

**Comment by wenkm** [comment]
怎么说呢，这玩意儿本身就是个函数，可以修改类的行为，可以当做注释来看，代码可读性更强？通过`@`食用，相当于 `Object.defineProperty`？

5大分类：
- 类
- 方法
- 方法参数
- 属性
- 访问器
 
应用场景：
- 在不改变原有类的情况下对其就行扩展
- 注释
- 装逼
